controller/engine_controller.go

```diff
@@ -560,22 +560,24 @@ func (ec *EngineController) GetInstance(obj interface{}) (*longhorn.InstanceProc
 	return c.ProcessGet(e.Name)
 }
 
-func (ec *EngineController) LogInstance(ctx context.Context, obj interface{}) (*imapi.LogStream, error) {
+func (ec *EngineController) LogInstance(ctx context.Context, obj interface{}) (*engineapi.InstanceManagerClient, *imapi.LogStream, error) {
```

Comment:
NIT: I am thinking if the InstanceManagerClient should be init by the caller of the interface rather than by itself. Returning the client looks weird...

Reply:
Agree the controller + handler stuff needs to be refactored, I didn't want to do this as part of this issue instead we should do the refactoring after we have done the proxy + monitor implementation since I don't think we will even need the handler abstraction afterwards.

I left a more details of this as part of the commit message.